designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/ScenarioStatistics.scala

```diff
@@ -0,0 +1,178 @@
+        CategoriesCount  -> categoriesCount,
+        VersionsMedian   -> versionsMedian,
+        VersionsAverage  -> versionsAverage,
+        VersionsMax      -> versionsMax,
+        VersionsMin      -> versionsMin,
+        AuthorsCount     -> authorsCount,
+        FragmentsMedian  -> fragmentsMedian,
+        FragmentsAverage -> fragmentsAverage
+      ) ++ uptimeStatsMap)
+        .map { case (k, v) => (k.toString, v.toString) }
+    }
+  }
+
+  def getActivityStatistics(
+      listOfActivities: List[DbProcessActivityRepository.ProcessActivity]
+  ): Map[String, String] = {
+    if (listOfActivities.isEmpty) {
+      Map.empty
+    } else {
+      //        Attachment stats
+      val sortedAttachmentCountList = listOfActivities.map(_.attachments.length).sorted
+      val attachmentAverage         = sortedAttachmentCountList.sum / sortedAttachmentCountList.length
+      val attachmentsTotal          = sortedAttachmentCountList.sum
+      //        Comment stats
+      val comments        = listOfActivities.map(_.comments.length).sorted
+      val commentsTotal   = comments.sum
+      val commentsAverage = comments.sum / comments.length
+
+      Map(
+        AttachmentsAverage -> attachmentAverage,
+        AttachmentsTotal   -> attachmentsTotal,
+        CommentsTotal      -> commentsTotal,
+        CommentsAverage    -> commentsAverage
+      ).map { case (k, v) => (k.toString, v.toString) }
+    }
+  }
+
+  def getComponentStatistic(componentList: List[component.ComponentListElement]): Map[String, String] = {
+    if (componentList.isEmpty) {
+      Map.empty
+    } else {
+      val withoutFragments      = componentList.filterNot(comp => comp.componentType == ComponentType.Fragment)
+      val componentsByNameCount = withoutFragments.groupBy(_.name).size
+
+      Map(
+        ComponentsCount -> componentsByNameCount
+      ).map { case (k, v) => (k.toString, v.toString) }
+    }
+  }
+
+  private val flinkDeploymentManagerType = DeploymentManagerType("flinkStreaming")
+
+  private val liteK8sDeploymentManagerType = DeploymentManagerType("lite-k8s")
+
+  private val liteEmbeddedDeploymentManagerType = DeploymentManagerType("lite-embedded")
+
+  private val knownDeploymentManagerTypes =
+    Set(flinkDeploymentManagerType, liteK8sDeploymentManagerType, liteEmbeddedDeploymentManagerType)
+
+  // We have four dimensions:
+  // - scenario / fragment
+  // - processing mode: streaming, r-r, batch
+  // - dm type: flink, k8s, embedded, custom
+  // - status: active (running), other
+  // We have two options:
+  // 1. To aggregate statistics for every combination - it give us 3*4*2 + 3*4 = 36 parameters
+  // 2. To aggregate statistics for every dimension separately - it gives us 2+3+4+1 = 10 parameters
+  // We decided to pick the 2nd option which gives a reasonable balance between amount of collected data and insights
+  private[statistics] def determineStatisticsForScenario(inputData: ScenarioStatisticsInputData): Map[String, Int] = {
```

Comment:
Maybe we could move the logic with combining results of this function here because here we have calculation for single data here and combining these results in the other place